launcher/minecraft/PackProfile.cpp

```diff
@@ -939,6 +942,65 @@ bool PackProfile::installCustomJar_internal(QString filepath)
     return true;
 }
 
+bool PackProfile::installAgents_internal(QStringList filepaths)
+{
+    // FIXME code duplication
+    const QString patchDir = FS::PathCombine(d->m_instance->instanceRoot(), "patches");
+    if (!FS::ensureFolderPathExists(patchDir))
+        return false;
+
+    const QString libDir = d->m_instance->getLocalLibraryPath();
+    if (!FS::ensureFolderPathExists(libDir))
+        return false;
+
+    for (const QString& source : filepaths) {
+        const QFileInfo sourceInfo(source);
+        const QString id = QUuid::createUuid().toString(QUuid::WithoutBraces);
+        const QString targetBaseName = id + ".jar";
+        const QString targetId = "org.prismlauncher.agent." + id;
+        const QString targetName = sourceInfo.completeBaseName() + " (agent)";
+        const QString target = FS::PathCombine(d->m_instance->getLocalLibraryPath(), targetBaseName);
+
+        const QFileInfo targetInfo(target);
+        if (targetInfo.exists())
```

Comment:
We are using UUID here, so we could just make this an assertion, as it *should* be "impossible".